### DISCUSSION TITLE
which key is good for a user leader key

### DISCUSSION THREAD

**Discussion by tkkcc** [helix-editor/helix#7822] [Q&A]
I used to use `,` as a leader key for shorten typing.

```txt
,e : :e<cr>
,x : :x<cr>
,q : :q!<cr>
,w : :w<cr>
,d : duplicate current line or selection
,D : duplicate current line or selection and comment out the previous one
...
```
then i found `,` and `;` already have meaningful mappings. while `space` is more like a community/official leader key, i don't want to mess with it. 

i believe a user leader key should be free(no existing mapping use it) and easy to type(one key)

is there any suggestion?

**Comment by pascalkuthe** [comment]
Basically the entire keymap is taken at this point. If you want to add a new submode then you need to move some default binding. 

Adding new things to the space menu night a better idea tough

**Reply by markstos** [reply]
It seems useful to have *one* key that reserved as a user-leader key.

I'm going to try combining the idea of using the "space" menu with the popular comma leader from Vim and use "space-comma" as a leader. Hopefully that will remain non-conflicting even as the space menu continues to evolve over time.

In `config.toml`, that means adding a section like this:

[keys.normal.space.","]

**Comment by gyreas** [comment]
Try `C-,` it's not bound on my setup, probably the same for yours. (It's also (maybe) convenient if you have Ctrl on Capslock)

**Comment by markstos** [comment]
It seems that semicolon: ";" is also available. It's bound in selection mode to collapse_selection, but does not appear bound in normal mode.

**Reply by the-mikedavis** [reply]
`;` is set in normal mode as well:

https://github.com/helix-editor/helix/blob/9c479e6d2de3bca9dec304f9182cee2b1c0ad766/helix-term/src/keymap/default.rs#L86

**Reply by quantonganh** [reply]
> `;` is set in normal mode as well:
> 
> https://github.com/helix-editor/helix/blob/9c479e6d2de3bca9dec304f9182cee2b1c0ad766/helix-term/src/keymap/default.rs#L86

What does `collapse_selection` really do in the normal mode?

**Reply by the-mikedavis** [reply]
It's the same behavior as in select mode - it takes any selection larger than a cursor and reduces it down to just a cursor.

**Comment by gyreas** [comment]
That's actually right. There's "-" too. So, you a number of options to explore. Prefix keys are truly awesome, they're like another keyboard layer
